react-with-json-logic/src/testCaseRunner/testCaseRunner.ts

```diff
@@ -0,0 +1,139 @@
+import { QuestionnaireEngine, Result } from "covquestions-js";
+import { Questionnaire, TestCase } from "covquestions-js/models/Questionnaire.generated";
+
+type TestResultSuccess = {
+  description: string;
+  success: true;
+};
+
+type TestResultError = {
+  description: string;
+  success: false;
+  errorMessage: string;
+};
+
+export type TestResult = TestResultSuccess | TestResultError;
+
+export function runTestCases(testQuestionnaire: Questionnaire): TestResult[] {
+  if (testQuestionnaire.testCases === undefined || testQuestionnaire.testCases.length === 0) {
+    return [];
+  }
+
+  return testQuestionnaire.testCases.map((testCase) => runOneTestCase(testQuestionnaire, testCase));
+}
+
+export function runOneTestCase(testQuestionnaire: Questionnaire, testCase: TestCase): TestResult {
+  const timeOfExecution = testCase.options?.fillInDate ? Date.parse(testCase.options?.fillInDate) / 1000 : undefined;
+
+  const engine = new QuestionnaireEngine(testQuestionnaire, timeOfExecution);
+
+  const questionCheck = checkQuestions(engine, testCase);
+  if (questionCheck) {
+    return questionCheck;
+  }
+
+  const resultsCheck = checkResults(engine.getResults(), testCase);
+  if (resultsCheck) {
+    return resultsCheck;
+  }
+
+  return { description: testCase.description, success: true };
+}
+
+function findUnusedElements(arrayWithPossibleUnusedElements: string[], otherArray: string[]): string[] {
+  if (otherArray.length !== arrayWithPossibleUnusedElements.length) {
+    return arrayWithPossibleUnusedElements.filter((it) => !otherArray.includes(it));
+  }
+
+  return [];
+}
+
+function checkQuestions(engine: QuestionnaireEngine, testCase: TestCase): TestResultError | undefined {
+  const { description } = testCase;
+  const questionMode = testCase.options?.questionMode;
+
+  const givenAnswers: string[] = [];
+
+  let question = engine.nextQuestion();
+
+  while (question !== undefined) {
+    const answerValue = testCase.answers[question.id];
```

Comment:
Is it possible to explicitly skip a question in a test case if `questionMode` is `strict`? Actually the value should be undefined then, right? Either because no answer was defined for the question or because the answer was explicitly set to undefined. But both options lead to `else` where an error will be returned in strict mode.
Maybe we should skip the strict check if the question is optional? Or introduce another property that defines if a question should be explicitly skipped?

Reply:
Good catch, I will think about it again. It worked before with the arrays instead of key-value pairs :). I will think about it again. Maybe it is okay to allow in strict mode to omit these answers for optional questions.